spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/data/SparkParquetReaders.java

```diff
@@ -497,6 +513,28 @@ protected MapData buildMap(ReusableMapData map) {
     }
   }
 
+  /** Variant reader to convert from Variant to Spark VariantVal */
+  private static class VariantReader
+      extends ParquetVariantReaders.DelegatingValueReader<Variant, VariantVal> {
```

Comment:
Style: I usually prefer to import the static class rather than use the full name.